scripts/deployer.ts

```diff
@@ -0,0 +1,279 @@
+import 'hardhat-deploy-ethers';
+import * as fs from 'fs';
+import { ContractReceipt, ContractTransaction } from 'ethers';
+import mainnetData from '../mainnet_deployed.json';
+import testnetData from '../testnet_deployed.json';
+import dummyData from '../example_testnet_deployed.json';
+import { getCurrentTimestamp } from 'hardhat/internal/hardhat-network/provider/utils/getCurrentTimestamp';
+import { ethers } from 'hardhat';
+import hre from 'hardhat';
+
+interface DeployedContract {
+  abi: Array<any>; // Improve these types
+  bytecode: string;
+  address: string;
+  block: number;
+  url: string;
+}
+
+interface DeployedData {
+  contracts: {
+    postageStamp: DeployedContract;
+    redistribution: DeployedContract;
+    staking: DeployedContract;
+    priceOracle: DeployedContract;
+    factory: DeployedContract;
+  };
+}
+
+let fileName = 'example_testnet_deployed.json';
+
+// URL
+const mainnetURL = 'https://gnosisscan.io/address/';
+const testnetURL = 'https://goerli.etherscan.io/address/';
+
+const blockChainVendor = hre.network.name;
+const networkID = hre.network.config.chainId;
+
+async function main(deployedData: DeployedData = dummyData) {
+  let contractData = {
+    addresses: {
+      postageStamp: '',
+      redistribution: '',
+      staking: '',
+      priceOracle: '',
+      factory: '',
+    },
+    blocks: {
+      postageStamp: 0,
+      redistribution: 0,
+      staking: 0,
+      priceOracle: 0,
+      factory: 0,
+    },
+  };
+
+  switch (blockChainVendor) {
+    case 'testnet':
+      contractData = await deployFactory(await JSON.parse(JSON.stringify(testnetData).toString()), contractData);
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      deployedData = testnetData;
+      break;
+    case 'mainnet':
+      contractData = await deployFactory(await JSON.parse(JSON.stringify(mainnetData).toString()), contractData);
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      deployedData = mainnetData;
+      break;
+    default:
+      contractData = await deployFactory(null, contractData);
+  }
+
+  if (!contractData.addresses.factory) {
+    handleErr('expected factory address');
+  }
+
+  contractData = await deployRedistribution(
+    await deployPriceOracle(await deployPostageStamp(await deployStaking(contractData)))
+  );
+
+  await rolesSetter(contractData);
+
+  await writeResult(deployedData, contractData);
+  await logResult(contractData);
+}
+
+async function deployFactory(deployed: any, contractData: any) {
+  if (deployed == null) {
+    const TestToken = await hre.ethers.getContractFactory('TestToken');
```

Comment:
Not sure about this, is this correct when deploying to the mainnet?

Reply:
yes, looks good

Reply:
@mrekucci is correct i believe, we should not deploy the testToken when deploying to mainnet

Reply:
instead we should fetch the BZZ token address from .env

Reply:
It'll be fetched from the `*_deployed.json` files once bootstrapped.